modules/spring-config/src/main/java/org/mule/runtime/config/internal/context/lazy/LazyMuleArtifactContext.java

```diff
@@ -288,29 +329,39 @@ public void initializeComponents(ComponentLocationFilter filter) {
     initializeComponents(filter, true);
   }
 
+  @Override
+  public void initializeAdditionalComponent(Location location) {
+    // TODO: check if we can remove this requirement
```

Comment:
Removing this requirement should be pretty easy. I'm not sure if we want to do it though.

Reply:
Done